fixbackend/workspaces/models/__init__.py

```diff
@@ -31,19 +31,29 @@ class Workspace:
     external_id: ExternalId
     owner_id: UserId
     members: List[UserId]
-    product_tier: ProductTier
+    selected_product_tier: ProductTier  # what the user currently selected in the UI
     created_at: datetime
     updated_at: datetime
     subscription_id: Optional[SubscriptionId] = None
     payment_on_hold_since: Optional[datetime] = None
+    highest_current_cycle_tier: Optional[ProductTier] = (
+        None  # which tier we saw as the highest until the end of the billing cycle
+    )
+    current_cycle_ends_at: Optional[datetime] = None  # when the active product tier ends, typically end of the month
+
+    def current_product_tier(self) -> ProductTier:
```

Comment:
Please add a comment describing:
- when to use selected product tier
- current product tier
- how the product tier is computed in billing